optimum/intel/openvino/configuration.py

```diff
@@ -259,6 +302,86 @@ def post_init(self):
                     f"For 8-bit quantization, `group_size` is expected to be set to -1, but was set to {self.group_size}"
                 )
 
+        if self.tokenizer is not None and not isinstance(self.tokenizer, str):
+            raise ValueError(f"Tokenizer is expected to be a string, but found {self.tokenizer}")
+
+
+@dataclass
+class OVQuantizationConfig(OVQuantizationConfigBase):
+    def __init__(
+        self,
+        ignored_scope: Optional[dict] = None,
+        num_samples: Optional[int] = 300,
+        preset: nncf.QuantizationPreset = None,
+        model_type: nncf.ModelType = nncf.ModelType.TRANSFORMER,
+        fast_bias_correction: bool = True,
+        overflow_fix: OverflowFix = OverflowFix.DISABLE,
+        weight_only: Optional[bool] = False,
+        **kwargs,
+    ):
+        """
+        Configuration class containing parameters related to model quantization with NNCF. Compared to weight
+        compression, during quantization both weights and activations are converted to lower precision.
+        For weight-only model quantization please see OVWeightQuantizationConfig.
+        Args:
+            ignored_scope (`dict`, *optional*):
+                An ignored scope that defines the list of model nodes to be ignored during quantization. Dictionary
+                entries provided via this argument are used to create an instance of `nncf.IgnoredScope` class.
+            num_samples (`int`, *optional*):
+                The maximum number of samples composing the calibration dataset.
+            preset (`nncf.QuantizationPreset`, *optional*):
+                A preset controls the quantization mode (symmetric and asymmetric).
+                It can take the following values:
+                - `performance`: Symmetric quantization of weights and activations.
+                - `mixed`: Symmetric quantization of weights and asymmetric quantization of activations.
+                Default value is None. In this case, `mixed` preset is used for `transformer`
+                model type otherwise `performance`.
+            model_type (`nncf.ModelType`, defaults to nncf.ModelType.TRANSFORMER):
+                Model type is needed to specify additional patterns in the model. Supported only `transformer` now.
+            fast_bias_correction (`bool`, defaults to True):
+                Whether to apply fast or full bias correction algorithm.
+            overflow_fix (`nncf.OverflowFix`, default to OverflowFix.DISABLE):
+                Parameter for controlling overflow fix setting.
+            weight_only (`bool`, *optional*):
+                Used to explicitly specify type of quantization (weight-only of full) to apply.
+        """
+        if weight_only is True:
+            logger.warning(
+                "Trying to create an instance of `OVQuantizationConfig` with `weight_only` being True. "
+                "Please check your configuration."
+            )
```

Comment:
It's just to make sure that nobody creates the config with the wrong `weight_only` property, e.g. `OVWeightQuantizationConfig(weight_only=False)`. Wrong property in turn will impact deserialization of quantization config: https://github.com/nikita-savelyevv/optimum-intel/blob/introduce-ov-quantization-config/optimum/intel/openvino/configuration.py#L152

Reply:
I remembered the reason this is needed. In the scenario when someone provides quantization config as dictionary instead of a Config object, there is some logic that tries to infer the type of the config based on the provided keys. However, this is not 100% reliable, so there's an additional flag to distinguish between weight-only and full quantization.

For example, if config is given as `dict(ignored_scope={"names": ["op_name"]})` than it's ambiguous which kind of quantization is intended because both kinds accept an ignored scope parameter. By default in such case `OVWeightQuantizationConfig` will be created and a warning [will be given](https://github.com/nikita-savelyevv/optimum-intel/blob/introduce-ov-quantization-config/optimum/intel/openvino/configuration.py#L146). 

To run full quantization with only ignored scope given, it's required to provide config as `dict(weight_only=False, ignored_scope={"names": ["op_name"]})`. In such case an instance of `OVQuantizationConfig` will be built and full quantization is run.

The warning you mention here happens when for example the config is given as `dict(bits=8, sym=True, weight_only=False)` which is confusing since parameters for weight only quantization are given, but `weight_only` hints at that the full quantization is intended. In such case, weight only quantization will be executed nevertheless, but the warning is given to avoid such confusion.

Reply:
I see, I think it would make sense to always create an instance of `OVWeightQuantizationConfig` by default and remove this parameter